app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/VoiceCommandsAdapter.kt

```diff
@@ -32,13 +37,16 @@ class VoiceCommandsAdapter(val items: ArrayList<String>, val context: Context?)
         val voiceCommand: TextView = view.voiceCommand
 
         init {
-            view.setOnClickListener {
-                val chatMessage = items[adapterPosition]
-                val splits = chatMessage.split("\n".toRegex()).dropLastWhile { it.isEmpty() }
-                val message = splits.joinToString(" ")
-                if (!chatMessage.isEmpty()) {
-                    chatPresenter.sendMessage(message, items[adapterPosition])
-                }
+            voiceCommand.setOnClickListener {
+                val message = items[adapterPosition]
+                val this_activity = context
+                (this_activity as ChatActivity).setText(message)
+                this_activity.searchChat.show()
+                this_activity.voiceSearchChat.show()
+                this_activity.fabsetting.show()
+                this_activity.btnSpeak.isEnabled = true
+                chatPresenter.startHotwordDetection()
+                this_activity.supportFragmentManager.popBackStackImmediate()
```

Comment:
Why not sending to presenter

Reply:
I tried that also. But, what I found that, as we are not starting the chatActivity (none of the following actions onResume, onPause, onCreate gets executed), so the presenter is failing to control any UI changes of the chatActivty. Hence, that method failed. Then, I found out this method useful.

Reply:
This doesn't seem to be a good way to set the click listener. To keep things clean, you should create and pass an item click listener (that requires adapter position and the message) from the view to the adapter, while initialising the adapter, and inside the init of view holder provide the adapter position and message to be set. Then, from your fragment, tell your presenter to start hotword detection and once that is done successfully, let the presenter tell the view to display any changes, that are required, to the user.

Reply:
@arundhati24 I even tried that method. Whats, the issue here is that when we do popBackStackImmediate(), the ChatActivity is opened. The issue here is, the state of ChatActivity is neither onPause, onResume or onCreate. So, even if I try to do anything using view, it simply fails. Those options that have onClick function in XML only work. Even, the chat presenter doesn't get access to the view in such cases(I have checked it). That was the main reason, I came up with this kind of solution, after so many tries. I also found, that the one who created the files of STT Fragment, also had a similar mindset.